Lib/test/test_logging.py

```diff
@@ -5755,6 +5755,16 @@ def test_compute_rollover_daily_attime(self):
         finally:
             rh.close()
 
+    def test_midnight_interval(self):
+        with warnings.catch_warnings(record=True) as w:
+            rh = logging.handlers.TimedRotatingFileHandler(
+                self.fn, encoding="utf-8", when='MIDNIGHT', interval=2, backupCount=0,
+                utc=True)
+            rh.close()
+        if w:
```

Comment:
Do you want the test to pass if no warning is raised? If not then in the above I would force deprecations on as exceptions via `warnings.simplefilter("error")`.

Reply:
Good point. Thanks for the review, Brett.

Reply:
Why not use `assertWarns()`?